.github/workflows/main.yml

```diff
@@ -0,0 +1,25 @@
+name: test
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+      - name: install
+        run: |
+          npm install -g npm
```

Comment:
why you need to install npm globally?

Reply:
TBH copied pasted this over from another repo where I wanted to make sure I have an up-to-date NPM, that will be used for subsequent installs

Reply:
Let me remove it, let's just use the one bundled with the Node version we use